tests/foreman/ui_airgun/test_discoveredhost.py

```diff
@@ -0,0 +1,414 @@
+    # Update default discovered host location
+    discovery_loc = entities.Setting().search(
+        query={'search': 'name="discovery_location"'})[0]
+    default_discovery_loc = discovery_loc.value
+    discovery_loc.value = loc.name
+    discovery_loc.update(['value'])
+    yield loc
+    discovery_loc.value = default_discovery_loc
+    discovery_loc.update(['value'])
+
+
+@fixture(scope='module')
+def provisioning_env(module_org, module_loc):
+    # Build PXE default template to get default PXE file
+    entities.ConfigTemplate().build_pxe_default()
+    return configure_provisioning(
+        org=module_org,
+        loc=module_loc,
+        os='Redhat {0}'.format(RHELRepository().repo_data['version']),
+    )
+
+
+@fixture
+def discovered_host():
+    return create_discovered_host()
+
+
+@fixture(scope='module')
+def module_host_group(module_org, module_loc):
+    host = entities.Host(organization=module_org, location=module_loc)
+    host.create_missing()
+    return entities.HostGroup(
+        organization=[module_org],
+        location=[module_loc],
+        medium=host.medium,
+        root_pass=gen_string('alpha'),
+        operatingsystem=host.operatingsystem,
+        ptable=host.ptable,
+        domain=host.domain,
+        architecture=host.architecture,
+    ).create()
+
+
+def _is_host_reachable(host, retries=12, iteration_sleep=5,
+                       expect_reachable=True):
+    """Helper to ensure given IP/hostname is reachable or not. The return value
+    returned depend from expect reachable value.
+
+    :param str host: The IP or hostname of host.
+    :param int retries: The polling retries.
+    :param int iteration_sleep: time to wait after each retry iteration.
+    :param bool expect_reachable: Whether we expect the host to be reachable.
+    :return bool
+    """
+    operator = '&&'
+    if not expect_reachable:
+        operator = '||'
+    cmd = ('for i in {{1..{0}}}; do ping -c1 {1} {2} exit 0; sleep {3};'
+           ' done; exit 1')
+    result = ssh.command(
+        cmd.format(retries, host, operator, iteration_sleep),
+        connection_timeout=30
+    )
+    if expect_reachable:
+        return not result.return_code
+    else:
+        return bool(result.return_code)
+
+
+@skip_if_not_set('compute_resources', 'vlan_networking')
+@tier3
+@upgrade
+def test_positive_pxe_based_discovery(
+        session, module_org, module_loc, provisioning_env):
+    """Discover a host via PXE boot by setting "proxy.type=proxy" in
+    PXE default
+
+    :id: 43a8857d-2f08-436e-97fb-ffec6a0c84dd
+
+    :Setup: Provisioning should be configured
+
+    :Steps: PXE boot a host/VM
+
+    :expectedresults: Host should be successfully discovered
+
+    :CaseLevel: System
+    """
+    with LibvirtGuest() as pxe_host:
+        host_name = pxe_host.guest_name
+        with session:
+            session.organization.select(org_name=module_org.name)
+            session.location.select(loc_name=module_loc.name)
```

Comment:
maybe we can implement something similar for `module_loc` as we have for `module_org` and remove all the statements